Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/RequirementNodeData.xml

```diff
@@ -6202,4 +6202,35 @@
     <CRequirementCountUpgrade id="AP_CountUpgradeAP_RoachVileCompleteOnlyTechTreeCheat">
         <Count Link="AP_RoachVile" State="CompleteOnly"/>
     </CRequirementCountUpgrade>
+    <CRequirementCountUpgrade id="AP_CountUpgradeAP_DisruptorPermanentCloakCompleteOnly">
+        <Flags index="TechTreeCheat" value="0"/>
+        <Count Link="AP_DisruptorPermanentCloak" State="CompleteOnly"/>
+    </CRequirementCountUpgrade>
+    <CRequirementCountUpgrade id="AP_CountUpgradeAP_DisruptorRestructuredThrustersCompleteOnly">
+        <Flags index="TechTreeCheat" value="0"/>
+        <Tooltip value="RequirementNode/Tooltip/##id##"/>
+        <Count Link="AP_DisruptorRestructuredThrusters" State="CompleteOnly"/>
+    </CRequirementCountUpgrade>
+    <CRequirementNot id="AP_NotCountUpgradeAP_DisruptorRestructuredThrustersCompleteOnly">
+        <Tooltip value="RequirementNode/Tooltip/##id##"/>
+        <OperandArray index="0" value="AP_CountUpgradeAP_DisruptorRestructuredThrustersCompleteOnly"/>
+    </CRequirementNot>
+    <CRequirementCountUpgrade id="AP_CountUpgradeAP_DisruptorPerfectedPowerCompleteOnly">
+        <Flags index="TechTreeCheat" value="0"/>
+        <Tooltip value="RequirementNode/Tooltip/##id##"/>
+        <Count Link="AP_DisruptorPerfectedPower" State="CompleteOnly"/>
+    </CRequirementCountUpgrade>
+    <CRequirementCountUpgrade id="AP_CountUpgradeAP_DisruptorRestrainedDestructionCompleteOnly">
+        <Flags index="TechTreeCheat" value="0"/>
+        <Tooltip value="RequirementNode/Tooltip/##id##"/>
+        <Count Link="AP_DisruptorRestrainedDestruction" State="CompleteOnly"/>
+    </CRequirementCountUpgrade>
+    <CRequirementNot id="AP_NotCountUpgradeAP_DisruptorPerfectedPowerCompleteOnly">
+        <Tooltip value="RequirementNode/Tooltip/##id##"/>
+        <OperandArray index="0" value="AP_CountUpgradeAP_DisruptorPerfectedPowerCompleteOnly"/>
+    </CRequirementNot>
+    <CRequirementNot id="AP_NotCountUpgradeAP_DisruptorRestrainedDestructionCompleteOnly">

```

Comment:
RequirementNodes have messed up ID formatting. Should be `AP_NotCountUpgradeDisruptorestrainedDestructionCompleteOnly`. Don't rely on the editor's ID generation, it's bad.